src/main/groovy/org/transmartproject/batch/concept/ConceptTree.groovy

```diff
@@ -40,24 +42,22 @@ class ConceptTree {
 
     void loadExisting(Collection<ConceptNode> nodes) {
         nodes.each { n ->
-            if (n.new) {
-                throw new IllegalArgumentException("Passed concept node is not expected to be new: $n")
-            }
             if (log.traceEnabled &&
                     nodeMap.containsKey(n.path)) {
                 log.trace "Replacing ${nodeMap[n.path]} with $n"
             }
 
             nodeMap[n.path] = n
+            savedNodes << n
         }
     }
 
-    Collection<ConceptNode> getNewConceptNodes() {
-        nodeMap.values().findAll { it.isNew() }
+    Set<ConceptNode> getNewConceptNodes() {
+        nodeMap.values().toSet() - savedNodes
```

Comment:
Avoid using the `-` operator on sets. It's very inefficient. 

    groovy -e '(1..50000 as Set) - (1..20000 as Set)'`

Takes 9 seconds on my machine, whereas:

    CLASSPATH=~/.m2/repository/com/google/guava/guava/18.0/guava-18.0.jar \
        time groovy -e 'println com.google.common.collect.Sets.difference((1..50000 as Set),(1..20000 as Set))'

Takes less than 1 (most of which is JVM startup).